### PR TITLE
feat: add sidebar drawer type

### PR DESCRIPTION
**Pull request by NoemiRozpara**
moved from https://github.com/react-navigation/navigation-ex/pull/301

### CODE REVIEW

**Comment by pincheira** [comment]
@NoemiRozpara is this compiling for you? I tried it out and it's got some double declaration error:

```
 web  Failed to compile.
./packages/drawer/src/views/DrawerView.tsx
SyntaxError: /Users/jp/Work/spare/permanent-sidebar/react-navigation/packages/drawer/src/views/DrawerView.tsx: Identifier 'handleDrawerOpen' has already been declared (158:8)

  156 |   }
  157 |
> 158 |   const handleDrawerOpen = () => {
      |         ^
  159 |     if (bigScreenSidebar) {
  160 |       return;
  161 |     }
```

Thank you 🙏

**Comment by NoemiRozpara** [comment]
Can you re-run the checks?

**Comment by osdnk** [comment]
rerunning

File: packages/drawer/src/views/Drawer.tsx [hunk 1]
```diff
@@ -92,6 +92,7 @@ type Props = {
   renderDrawerContent: Renderer;
   renderSceneContent: Renderer;
   gestureHandlerProps?: React.ComponentProps<typeof PanGestureHandler>;
+  isBigScreen?: boolean;
```
Comment (satya164) [inline packages/drawer/src/views/Drawer.tsx]: We don't need a `isBigScreen` prop. We can pass something like `drawerType={screenWidth > 800 ? 'sidebar' : 'front'}`

File: packages/drawer/src/views/Drawer.tsx [hunk 2]
```diff
@@ -601,35 +610,59 @@ export default class DrawerView extends React.PureComponent<Props> {
               importantForAccessibility={open ? 'no-hide-descendants' : 'auto'}
               style={styles.content}
             >
-              {renderSceneContent({ progress: this.progress })}
+              {renderSceneContent({
+                // make 'sidebar' initially visible on the big screen to avoid annoying animation
+                progress: this.bigScreenSidebar
+                  ? new Animated.Value(1)
```
Comment (satya164) [inline packages/drawer/src/views/Drawer.tsx]: nit: lets move this to top of the scope to avoid creating this animated value on render

File: packages/drawer/src/views/Drawer.tsx [hunk 3]
```diff
@@ -195,6 +197,9 @@ export default class DrawerView extends React.PureComponent<Props> {
     }
   };
 
+  private bigScreenSidebar =
```
Comment (satya164) [inline packages/drawer/src/views/Drawer.tsx]: we need to move it to render so that it respects prop changes

File: packages/drawer/src/views/Drawer.tsx [hunk 4]
```diff
@@ -214,11 +219,12 @@ export default class DrawerView extends React.PureComponent<Props> {
...
   private gestureX = new Value<number>(0);
   private offsetX = new Value<number>(0);
-  private position = new Value<number>(0);
+  private position = new Value<number>(this.bigScreenSidebar ? 1 : 0);
 
   private containerWidth = new Value<number>(0);
   private drawerWidth = new Value<number>(0);
-  private drawerOpacity = new Value<number>(0);
+  // make drawer initially visible on the big screen to avoid annoying animation
+  private drawerOpacity = new Value<number>(this.bigScreenSidebar ? 1 : 0);
```
Comment (satya164) [inline packages/drawer/src/views/Drawer.tsx]: I think we can just to `opacity: drawerType === 'sidebar' : 1 : this.drawerOpacity` below

File: packages/drawer/src/views/Drawer.tsx [hunk 5]
```diff
@@ -214,11 +219,12 @@ export default class DrawerView extends React.PureComponent<Props> {
   private velocityX = new Value<number>(0);
   private gestureX = new Value<number>(0);
   private offsetX = new Value<number>(0);
-  private position = new Value<number>(0);
+  private position = new Value<number>(this.bigScreenSidebar ? 1 : 0);
```
Comment (satya164) [inline packages/drawer/src/views/Drawer.tsx]: Probably don't need to adjust `position`. We could probably hardcode the value in render if we need it

File: packages/drawer/src/views/Drawer.tsx [hunk 6]
```diff
@@ -601,35 +610,59 @@ export default class DrawerView extends React.PureComponent<Props> {
...
           <Animated.View
             accessibilityViewIsModal={open}
             removeClippedSubviews={Platform.OS !== 'ios'}
             onLayout={this.handleDrawerLayout}
             style={[
               styles.container,
-              isRight ? { right: offset } : { left: offset },
-              {
-                transform: [{ translateX: drawerTranslateX }],
+              !this.bigScreenSidebar && {
```
Comment (satya164) [inline packages/drawer/src/views/Drawer.tsx]: Let's keep the style in a style object in `StyleSheet.create` and use it here

File: packages/drawer/src/views/DrawerView.tsx [hunk 7]
```diff
@@ -88,8 +88,12 @@ export default function DrawerView({
   sceneContainerStyle,
 }: Props) {
   const [loaded, setLoaded] = React.useState([state.index]);
+
+  const { width, height } = Dimensions.get('window');
+  const isBigScreen = Math.min(width, height) >= 768;
```
Comment (satya164) [inline packages/drawer/src/views/DrawerView.tsx]: I think maybe we can leave the automatic switching for now since now it's stable and this can be considered a breaking change. Let the user pass the `drawerType` explicitly

File: packages/drawer/src/types.tsx [hunk 8]
```diff
@@ -27,7 +27,7 @@ export type DrawerNavigationConfig<T = DrawerContentOptions> = {
    * - `back`: The drawer is revealed behind the screen on swipe.
    * - `slide`: Both the screen and the drawer slide on swipe to reveal the drawer.
    */
-  drawerType?: 'front' | 'back' | 'slide';
+  drawerType?: 'front' | 'back' | 'slide' | 'sidebar';
```
Comment (satya164) [inline packages/drawer/src/types.tsx]: Let's change to `'sidebar'` to `'permanent'` according to @EvanBacon's suggestion

```suggestion
  drawerType?: 'front' | 'back' | 'slide' | 'permanent';
```

File: packages/stack/src/TransitionConfigs/CardStyleInterpolators.tsx [hunk 9]
```diff
@@ -1,4 +1,5 @@
 import { Animated } from 'react-native';
+import { isIphoneX } from 'react-native-iphone-x-helper';
```
Comment (satya164) [inline packages/stack/src/TransitionConfigs/CardStyleInterpolators.tsx]: This change seems unrelated

**Comment by satya164** [comment]
Merged in https://github.com/react-navigation/react-navigation/commit/6a5d0a035afae60d91aef78401ec8826295746fe
